receiver/splunkhecreceiver/receiver.go

```diff
@@ -51,7 +51,7 @@ const (
 	responseErrInternalServerError    = "Internal Server Error"
 	responseErrUnsupportedMetricEvent = "Unsupported metric event"
 	responseErrUnsupportedLogEvent    = "Unsupported log event"
-
+	responseErrHandlingIndexedFields  = `{"text":"Error in handling indexed fields","code":15,"invalid-event-number":%d}`
```

Comment:
Is there any reason to have a JSON formatted string here with substitution?

Reply:
Actually, splunk returns the message in JSON format that I have used for `responseErrHandlingIndexedFields`. And `invalid-event-number` points to the index of the event where the problem is.

`failRequest` method sets `Content-Type` as `application/json` but response body is raw text.
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/c3d68f6a8ff1affc07d590ebc4cac6ca95768543/receiver/splunkhecreceiver/receiver.go#L425

We also need to change other err messages to JSON format with respect to the actual splunk response. I'll create a separate issue for that.

Reply:
Huh, interesting. Thanks for checking that for me.

Reply:
Created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18097